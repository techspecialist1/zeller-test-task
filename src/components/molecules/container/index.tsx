import React, {FC} from 'react';
import {SafeAreaView, View} from 'react-native';

import {styles} from './styles';

type ContainerProps = {
  children: React.ReactNode;
};

const Container: FC<ContainerProps> = ({children}) => {
  return (
    <SafeAreaView style={styles.safeArea}>
      <View style={styles.container}>{children}</View>
    </SafeAreaView>
  );
};

export default Container;
