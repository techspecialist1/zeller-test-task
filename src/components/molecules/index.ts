import ListView from './listView';
import Container from './container';
import RadioButton from '../atoms/radioButton';
import SearchInput from '../atoms/searchInput';
import ActivityIndicator from '../atoms/activityIndicator';
import ErrorBox from '../atoms/errorBox';

export {
  ListView,
  ActivityIndicator,
  RadioButton,
  SearchInput,
  Container,
  ErrorBox,
};
