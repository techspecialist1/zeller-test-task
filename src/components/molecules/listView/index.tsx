import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';

import {styles} from './style';

type ListViewProps = {
  item: {
    name: string;
    role: string;
  };
  onPress: () => void;
};

const ListView: React.FC<ListViewProps> = props => {
  const {item, onPress} = props;

  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.userList}>
        <View style={styles.profile}>
          <Text style={styles.profileText}>{item?.name?.slice(0, 1)}</Text>
        </View>
        <View style={styles.listName}>
          <Text style={styles.name}>{item?.name}</Text>
          <Text style={styles.role}>{item?.role}</Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default ListView;
