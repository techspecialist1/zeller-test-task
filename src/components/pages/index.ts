import HomePage from './homePage';
import DetailPage from './detailPage';

export {HomePage, DetailPage};
