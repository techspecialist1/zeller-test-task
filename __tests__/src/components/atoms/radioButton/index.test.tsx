import React from 'react';
import renderer from 'react-test-renderer';
import {Text, TouchableOpacity} from 'react-native';
import {RadioButton} from '../../../../../src/components/molecules';

describe('RadioButton', () => {
  const mockOnChange = jest.fn();

  const defaultProps = {
    label: 'Option 1',
    onChange: mockOnChange,
    activeButton: 'Option 1',
    styles: {
      activeButton: {},
      inActiveButton: {},
      radio: {},
    },
  };

  test('renders correctly with default props', () => {
    const component = renderer.create(<RadioButton {...defaultProps} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('calls onChange function when pressed', () => {
    const component = renderer.create(<RadioButton {...defaultProps} />);
    const instance = component.root;

    const touchableOpacity = instance.findByType(TouchableOpacity);
    touchableOpacity.props.onPress();

    expect(mockOnChange).toHaveBeenCalledWith('Option 1');
  });

  test('renders label correctly', () => {
    const component = renderer.create(<RadioButton {...defaultProps} />);
    const instance = component.root;

    const label = instance.findByType(Text);

    expect(label.props.children).toBe('Option 1');
  });
});
