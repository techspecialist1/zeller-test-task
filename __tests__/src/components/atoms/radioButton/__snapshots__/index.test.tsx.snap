// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#E8F2FB",
      "borderRadius": 6.153846153846153,
      "flexDirection": "row",
      "opacity": 1,
      "paddingLeft": 9.846153846153847,
      "paddingVertical": 9.846153846153847,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "#0171CD",
          "borderRadius": 12.307692307692307,
          "borderWidth": 1.2307692307692308,
          "height": 24.615384615384613,
          "justifyContent": "center",
          "width": 24.615384615384613,
        },
        null,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#0171CD",
            "borderRadius": 7.384615384615385,
            "height": 14.76923076923077,
            "width": 14.76923076923077,
          },
          null,
        ]
      }
    />
  </View>
  <View
    style={
      {
        "alignSelf": "center",
        "paddingLeft": 12.307692307692307,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000",
          "textTransform": "capitalize",
        }
      }
    >
      Option 1
    </Text>
  </View>
</View>
`;
